activejob/test/cases/translation_test.rb

```diff
@@ -0,0 +1,20 @@
+require 'helper'
+require 'jobs/translated_hello_job'
+
+class TranslationTest < ActiveSupport::TestCase
+  setup do
+    JobBuffer.clear
+    I18n.available_locales = [:en, :de]
+    @job = TranslatedHelloJob.new('Johannes')
+  end
+
+  teardown do
+    I18n.available_locales = [:en]
+  end
+
+  test 'it performs the job in the given locale' do
+    @job.locale = :de
+    @job.perform_now
```

Comment:
@robin850 in the regular tests adapters are set to inline mode so when perform_later is called the adapters are executing the job immediately. SP probably still forks in inline mode. We did had some integration tests which starts adapters workers and I would test in there that the locale is set